### DISCUSSION TITLE
How to use svg in ebiten

### DISCUSSION THREAD

**Discussion by sanguinedab** [hajimehoshi/ebiten#3177] [Q&A]
Hello, I want to use svg in this way. `img, _, err := ebitenutil.NewImageFromFileSystem(resources.ImagesFS, path) ` （resources.ImagesFS embed svg file），Is it possible? Or what kind of method should I use? I am a newbie in using svg.

**Comment by hajimehoshi** [comment]
SVG is not available in Ebitengine so far unfortunately.

**Reply by sanguinedab** [reply]
> SVG is not available in Ebitengine so far unfortunately.

Okay, actually what I want to solve is that my PNG images will appear jagged on small screens such as mobile devices. Do you have any good suggestions? Use images with a larger resolution?

**Reply by hajimehoshi** [reply]
I'd use a large resolution image as a source and render it with shrinking [`GeoM`](https://pkg.go.dev/github.com/hajimehoshi/ebiten/v2#GeoM) and [`FilterLinear`](https://pkg.go.dev/github.com/hajimehoshi/ebiten/v2#FilterLinear).

**Reply by sanguinedab** [reply]
> I'd use a large resolution image as a source and render it with shrinking [`GeoM`](https://pkg.go.dev/github.com/hajimehoshi/ebiten/v2#GeoM) and [`FilterLinear`](https://pkg.go.dev/github.com/hajimehoshi/ebiten/v2#FilterLinear).

Thank you, I will try it right away. Are there any plans to support svg in the future?

**Reply by hajimehoshi** [reply]
> Are there any plans to support svg in the future?

No so far, since 1) it is pretty hard to support a lot of features of SVGs and 2) it is hard to assure the rendering quality with the current Ebitengine implimentation with GPU utilization. I think most SVG rendering is actually done by CPU, not GPU.

**Reply by tinne26** [reply]
If we are talking about static SVG images that you don't need to resize dynamically, you can totally use a golang SVG library to render it to a regular image.Image, and then load it to Ebitengine with NewImageFromImage. tdewolff/canvas is an example of such a library. Using SVG is not a problem in Ebitengine, only dynamically scaled SVG is a problem.

**Reply by sanguinedab** [reply]
> If we are talking about static SVG images that you don't need to resize dynamically, you can totally use a golang SVG library to render it to a regular image.Image, and then load it to Ebitengine with NewImageFromImage. tdewolff/canvas is an example of such a library. Using SVG is not a problem in Ebitengine, only dynamically scaled SVG is a problem.

Thank you very much. I can use svg in this way. The reason why I want to use svg is that the image may become jagged after scaling on different devices. I will test it in detail. At present, it seems to have the same effect as using png. almost
`

	lenna, err := os.Open("example.svg")
	if err != nil {
		panic(err)
	}

	// Create a new image object
	c, err := canvas.ParseSVG(lenna)
	if err != nil {
		panic(err)
	}

	canvasImg := rasterizer.Draw(c, canvas.DPMM(96.0/25.4), canvas.DefaultColorSpace)
	img := ebiten.NewImageFromImage(canvasImg)
	t.Logf("img: %v", img.Bounds())

`

**Reply by superloach** [reply]
In the end Ebitengine still only works with rasterised images - you'd want to rasterise the SVG at the highest scale you'll want it, and then downscale as needed, perhaps using multiple scales to avoid artifacts. Doing this from SVG data at startup is fine, but as your assets grow plentiful you may also want to do your rasterisation ahead-of-time.
